elogia_project/models/project_task.py

```diff
@@ -10,6 +10,172 @@
+                        list_error.append({'error': 'contract_valid', 'item': slot})
+                    self.remove_slot(slot, list_ok)
+                if contract.date_start > init_date.date():
+                    if slot.id not in self.check_any_value(list_error):
+                        list_error.append({'error': 'date_init', 'item': slot})
+                    self.remove_slot(slot, list_ok)
+                if contract.date_end < end_date.date() and next_month > contract.date_end:
+                    if slot.id not in self.check_any_value(list_error):
+                        list_error.append({'error': 'date_end', 'item': slot})
+                    self.remove_slot(slot, list_ok)
+            else:
+                if slot.id not in self.check_any_value(list_error):
+                    list_error.append({'error': 'not_contract', 'item': slot})
+                self.remove_slot(slot, list_ok)
+            slot_filter_duple = view_slot_origin.filtered(lambda e: e.end_datetime.date() == next_month)
+            if slot_filter_duple:
+                if slot.id not in self.check_any_value(list_error):
+                    list_error.append({'error': 'duplicity', 'item': slot})
+                self.remove_slot(slot, list_ok)
+            if any(employee for employee in leave_filter.mapped('employee_ids') if leave_filter and employee == slot.employee_id):
+                if slot.id not in self.check_any_value(list_error):
+                    list_error.append({'error': 'time', 'item': slot})
+                self.remove_slot(slot, list_ok)
+            if slot.project_id:
+                contact = slot.employee_id.address_home_id
+                if contact.id not in slot.project_id.message_follower_ids.mapped('partner_id').ids:
+                    if slot.id not in self.check_any_value(list_error):
+                        list_error.append({'error': 'not_project', 'item': slot})
+                    self.remove_slot(slot, list_ok)
+                if slot.project_id.date < next_month:
+                    if slot.id not in self.check_any_value(list_error):
+                        list_error.append({'error': 'not_project_date', 'item': slot})
+                    self.remove_slot(slot, list_ok)
+            if slot.task_id:
+                user = slot.employee_id.user_id
+                if user.id not in slot.task_id.user_ids.ids:
+                    if slot.id not in self.check_any_value(list_error):
+                        list_error.append({'error': 'not_task', 'item': slot})
+                    self.remove_slot(slot, list_ok)
+                if slot.task_id.date_deadline < next_month:
+                    if slot.id not in self.check_any_value(list_error):
+                        list_error.append({'error': 'not_task_date', 'item': slot})
+                    self.remove_slot(slot, list_ok)
+        if list_ok:
+            slot_filters = view_slot.filtered(lambda e: e.id in list_ok)
+        if list_error:
```

Comment:
si luego recorres list_error con el for, no es necesesario prguntar  if list_error:

Reply:
Oído